test/e2e/vsphere_volume_placement.go

```diff
@@ -28,209 +32,355 @@ import (
-			deletePodAndWaitForVolumeToDetach(c, ns, vsp, node1Name, pod, volumePath)
+		// Create empty files on the mounted volumes on the pod to verify volume is writable
+		// Verify newly and previously created files present on the volume mounted on the pod
+		volumeFiles := []string{
+			fmt.Sprintf("/mnt/volume1/%v_1.txt", ns),
+			fmt.Sprintf("/mnt/volume2/%v_1.txt", ns),
+		}
+		createAndVerifyFilesOnVolume(ns, pod.Name, volumeFiles, volumeFiles)
+		deletePodAndWaitForVolumeToDetach(f, c, pod, vsp, node1Name, volumePaths)
+
+		By(fmt.Sprintf("Creating pod on the node: %v with volume :%v  and volume: %v", node1Name, volumePaths[0], volumePaths[1]))
+		pod = createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, volumePaths)
+		// Create empty files on the mounted volumes on the pod to verify volume is writable
+		// Verify newly and previously created files present on the volume mounted on the pod
+		newEmptyFileNames := []string{
+			fmt.Sprintf("/mnt/volume1/%v_2.txt", ns),
+			fmt.Sprintf("/mnt/volume2/%v_2.txt", ns),
+		}
+		volumeFiles = append(volumeFiles, newEmptyFileNames[0])
+		volumeFiles = append(volumeFiles, newEmptyFileNames[1])
+		createAndVerifyFilesOnVolume(ns, pod.Name, newEmptyFileNames, volumeFiles)
+		deletePodAndWaitForVolumeToDetach(f, c, pod, vsp, node1Name, volumePaths)
+	})
 
-			By("Creating pod on the another node: " + node2Name)
-			pod = createPodWithVolumeAndNodeSelector(c, ns, vsp, node2Name, node2KeyValueLabel, volumePath)
-			deletePodAndWaitForVolumeToDetach(c, ns, vsp, node2Name, pod, volumePath)
-		})
+	/*
+		Test Back-to-back pod creation/deletion with different volume sources on the same worker node
+		    1. Create volumes - vmdk2
+		    2. Create pod Spec - pod-SpecA with volume path of vmdk1 and NodeSelector set to label assigned to node1.
+		    3. Create pod Spec - pod-SpecB with volume path of vmdk2 and NodeSelector set to label assigned to node1.
+		    4. Create pod-A using pod-SpecA and wait for pod to become ready.
+		    5. Create pod-B using pod-SpecB and wait for POD to become ready.
+		    6. Verify volumes are attached to the node.
+		    7. Create empty file on the volume to make sure volume is accessible. (Perform this step on pod-A and pod-B)
+		    8. Verify file created in step 5 is present on the volume. (perform this step on pod-A and pod-B)
+		    9. Delete pod-A and pod-B
+		    10. Repeatedly (5 times) perform step 4 to 9 and verify associated volume's content is matching.
+		    11. Wait for vmdk1 and vmdk2 to be detached from node.
+	*/
+	It("test back to back pod creation and deletion with different volume sources on the same worker node", func() {
+		var (
+			podA                *v1.Pod
+			podB                *v1.Pod
+			testvolumePathsPodA []string
+			testvolumePathsPodB []string
+			podAFiles           []string
+			podBFiles           []string
+		)
+
+		defer func() {
+			By("clean up undeleted pods")
+			framework.DeletePodWithWait(f, c, podA)
```

Comment:
Test just creates two pods - podA and podB. 
`podAFiles` and `podBFiles` contains set of empty files created in the volumes mounted on these pods. used for verification of the volume content.